src/Sign.Core/SignatureProviders/ClickOnceSignatureProvider.cs

```diff
@@ -129,19 +129,26 @@ public async Task SignAsync(IEnumerable<FileInfo> files, SignOptions options)
                             throw new Exception(message);
                         }
 
-                        // Read the publisher name from the manifest for use below
-                        XDocument manifestDoc = XDocument.Load(manifestFile.FullName);
-                        XNamespace ns = manifestDoc.Root!.GetDefaultNamespace();
-                        XElement? publisherIdentity = manifestDoc.Root.Element(ns + "publisherIdentity");
-                        string publisherName = publisherIdentity!.Attribute("name")!.Value;
+                        string publisherParam = "";
 
-                        var publisherParam = "";
-
-                        Dictionary<string, List<string>> dict = DistinguishedNameParser.Parse(publisherName);
-                        if (dict.TryGetValue("CN", out var cns))
+                        if (string.IsNullOrEmpty(options.PublisherName))
+                        {
+                            // Read the publisher name from the manifest for use below
+                            XDocument manifestDoc = XDocument.Load(manifestFile.FullName);
+                            XNamespace ns = manifestDoc.Root!.GetDefaultNamespace();
+                            XElement? publisherIdentity = manifestDoc.Root.Element(ns + "publisherIdentity");
+                            string publisherName = publisherIdentity!.Attribute("name")!.Value;
+                            Dictionary<string, List<string>> dict = DistinguishedNameParser.Parse(publisherName);
+ 
+                            if (dict.TryGetValue("CN", out var cns))
+                            {
+                                // get the CN. it may be quoted
+                                publisherParam = $@"-pub ""{string.Join("+", cns.Select(s => s.Replace("\"", "")))}"" ";
```

Comment:
Do we have test for this code path?

Reply:
Yep, right [here](https://github.com/dotnet/sign/pull/568/files#diff-8c821ccfd0139e3ab038bc0f20d592d1899314a0b7be71caf9e63860fa1f94f8R199).